### PR TITLE
i#6681: Fix Doxygen file generation for GitHub pages.

### PR DESCRIPTION
**Pull request by brettcoon**
Doxygen 1.9.8 changed the way javascript menu files are generated, by inlining much of the menu hierarchy that was previously written to a separate file per page.

Updates cmake rules to handle the new javascript.  Also removes extra curly braces from a doxygen page that were inadvertently triggering GitHub's Liquid template expansion and removing the intended text.

This fix was verified by building the embedded doxygen files, copying them to a
local dynamorio.github.io workspace, running a Jekyll server to host the pages,
and comparing the local site vs. the online dynamorio.org pages.

Issue: #6681

### CODE REVIEW

File: api/docs/CMake_rundoxygen.cmake [hunk 1]
```diff
@@ -256,6 +256,11 @@ if (embeddable)
   endforeach ()
```
Comment (derekbruening) [inline api/docs/CMake_rundoxygen.cmake:256]: > previously written to page_user_docs.js,

nit: It was written to a file per page: not just to this one particular page of ours.

File: api/docs/CMake_rundoxygen.cmake [hunk 2]
```diff
@@ -286,7 +291,31 @@ if (embeddable)
...
+          message(FATAL_ERROR "Cannot move menu entry for \"Extension API\"")
+        endif ()
+        # Now insert the Extension API menu right above Disassembly Library.
+        if (NOT string MATCHES "Disassembly Library")
+          message(FATAL_ERROR "Cannot find menu entry for page \"Disassembly Library\"")
+        endif ()
+        string(REGEX REPLACE "(\n[^\n]+\"Disassembly Library)" "${ext_entry}\\1"
+          string "${string}")
+        # Set the user_docs flag since we found its contents here.
+        set(found_user_docs ON)
```
Comment (derekbruening) [inline api/docs/CMake_rundoxygen.cmake:314]: Add a sanity check !found_user_docs down on line 349 where it finds page_user_docs.js?

File: api/docs/CMake_rundoxygen.cmake (hunk 1 above)
Comment (derekbruening) [inline api/docs/CMake_rundoxygen.cmake:256]: Describe briefly how this was tested?  I.e., that you actually did do a local jekyll check and aren't just guessing that the result looks right.

**Review by derekbruening — APPROVED** [review]
Thank you for the fix!

File: api/docs/CMake_rundoxygen.cmake (hunk 2 above)
Comment (brettcoon) [reply api/docs/CMake_rundoxygen.cmake:314]: Done.

File: api/docs/CMake_rundoxygen.cmake (hunk 1 above)
Comment (brettcoon) [reply api/docs/CMake_rundoxygen.cmake:256]: Updated the PR description.

**Pull request merged**